public/app/features/alerting/unified/components/rule-editor/CloudAlertPreview.tsx

```diff
@@ -0,0 +1,131 @@
+import { css } from '@emotion/css';
+import { max } from 'lodash';
+import React from 'react';
+
+import { DataFrame, GrafanaTheme2 } from '@grafana/data/src';
+import { Icon, TagList, Tooltip, useStyles2 } from '@grafana/ui/src';
+
+import { GrafanaAlertState, isGrafanaAlertState, Labels } from '../../../../../types/unified-alerting-dto';
+import { AlertStateTag } from '../rules/AlertStateTag';
+
+interface AlertPreviewInstance {
+  state: GrafanaAlertState;
+  info?: string;
+  labels: Labels;
+}
+
+interface AlertPreview {
+  instances: AlertPreviewInstance[];
+}
+
+function mapDataFrameToAlertPreview({ fields }: DataFrame): AlertPreview {
+  const labelFields = fields.filter((field) => !['State', 'Info'].includes(field.name));
+  const stateFieldIndex = fields.findIndex((field) => field.name === 'State');
+  const infoFieldIndex = fields.findIndex((field) => field.name === 'Info');
+
+  const labelIndexes = labelFields.map((labelField) => fields.indexOf(labelField));
+
+  const maxValues = max(fields.map((f) => f.values.length)) ?? 0;
+
+  const instances: AlertPreviewInstance[] = [];
+
+  for (let index = 0; index < maxValues; index++) {
+    const labelValues = labelIndexes.map((labelIndex) => [
+      fields[labelIndex].name,
+      fields[labelIndex].values.get(index),
+    ]);
+    const state = fields[stateFieldIndex].values.get(index);
+    const info = fields[infoFieldIndex].values.get(index);
+
+    if (labelValues.length > 0 && isGrafanaAlertState(state)) {
```

Comment:
I think if we add a condition constant with a descriptive name could help us to remember what are we checking in here. For example:
`const canWeAddInstance =  labelValues.length > 0 && isGrafanaAlertState(state)`